### PR TITLE
docs(contributing): add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by robinjoseph08**
Since we've had more people contributing to this repo, I thought it would be helpful to list all the things we desire in open-source PRs. This should cut down on the feedback loop for when people submit these PRs.

### CODE REVIEW

**Comment by mgartner** [comment]
lgtm

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Contributing
+
+We encourage any form of contribution, whether that be issues, comments, or pull requests. If you are going to be submitting a PR, there are a few things we would appreciate that you do to keep the codebase clean:
```
Comment (elnaz) [inline CONTRIBUTING.md]: For the last sentence, what do you think of `If you plan to submit a Pull Request, please follow these guidelines to help keep the codebase clean`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+* **Write tests.** We enforce 100% code coverage on this repo so any new code that gets written should have accompanying tests.
```
Comment (elnaz) [inline CONTRIBUTING.md]: `any new code should be accompanied by tests`?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+* **Follow the linter.** We use our [ESLint configuration](https://github.com/lob/eslint-config-lob), and we run `npm run lint` in our Travis builds.
+* **Follow the commit pattern.** We use [`generate-changelog`](https://github.com/lob/generate-changelog) for automatic changelog generation whenever we deploy, so please follow the commit message format: `feat(paths): add glob support`.
```
Comment (elnaz) [inline CONTRIBUTING.md]: `whenever we publish a new version` instead of  `whenever we deploy`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+* **Squash your commits.** Ideally, your PR has one commit in it. If you you need to add another one because of a typo or refactor, you should squash your commits together. This makes the changelog and commit history much more managable to go through. If you need help squashing them, we can always help you out!
```
Comment (elnaz) [inline CONTRIBUTING.md]: `your PR should have one commit in it`.

**Comment by elnaz** [comment]
LGTM, just a few suggestions

**Comment by robinjoseph08** [comment]
@elnaz updated and merging!

**Pull request merged**